src/SprykerEco/Zed/Unzer/Persistence/UnzerEntityManager.php

```diff
@@ -0,0 +1,156 @@
+<?php
+
+/**
+ * MIT License
+ * For full license information, please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace SprykerEco\Zed\Unzer\Persistence;
+
+use Generated\Shared\Transfer\MerchantUnzerParticipantTransfer;
+use Generated\Shared\Transfer\PaymentUnzerOrderItemTransfer;
+use Generated\Shared\Transfer\PaymentUnzerTransactionTransfer;
+use Generated\Shared\Transfer\PaymentUnzerTransfer;
+use Orm\Zed\Unzer\Persistence\SpyMerchantUnzerParticipant;
+use Spryker\Zed\Kernel\Persistence\AbstractEntityManager;
+use SprykerEco\Zed\Unzer\Persistence\Mapper\UnzerPersistenceMapper;
+
+/**
+ * @method \SprykerEco\Zed\Unzer\Persistence\UnzerPersistenceFactory getFactory()
+ */
+class UnzerEntityManager extends AbstractEntityManager implements UnzerEntityManagerInterface
+{
+    /**
+     * @param \Generated\Shared\Transfer\PaymentUnzerTransfer $paymentUnzerTransfer
+     *
+     * @return \Generated\Shared\Transfer\PaymentUnzerTransfer
+     */
+    public function savePaymentUnzerEntity(PaymentUnzerTransfer $paymentUnzerTransfer): PaymentUnzerTransfer
+    {
+        $paymentUnzerEntity = $this->getFactory()
+            ->createPaymentUnzerQuery()
+            ->filterByFkSalesOrder($paymentUnzerTransfer->getIdSalesOrder())
+            ->filterByOrderId($paymentUnzerTransfer->getOrderId())
+            ->findOneOrCreate();
+
+        $paymentUnzerEntity = $this->getMapper()
+            ->mapPaymentUnzerTransferToPaymentUnzerEntity($paymentUnzerTransfer, $paymentUnzerEntity);
+        $paymentUnzerEntity->save();
+
+        return $this->getMapper()
```

Comment:
Maybe it will be better to introduce a variable for a mapper and not create it twice?

Reply:
Same for other methods here.